syntax = "proto3";
import "google/protobuf/empty.proto";

package Client_Server;

// Mensaje para enviar información del cliente.
message ClientInfo {
  string name = 1;
  string ip = 2;
  int32 port = 3;
}

message ClientName {
    string name = 1;
}

// Respuesta del servidor.
message RegisterResponse {
  bool success = 1;
  string message = 2;
}

// Mensaje para devolver parámetros del cliente.
message ClientParams {
  string ip = 1;
  int32 port = 2;
}

// Definición del servicio de registro.
service RedisService {
  rpc RegisterClient(ClientInfo) returns (RegisterResponse);
  rpc GiveParameters(ClientName) returns (ClientParams);
}
